### ISSUE TITLE
[MJAVADOC-76] doc-files/* not copied if project does not have modules

### ISSUE THREAD

**Issue by olamy** [olamy/maven-javadoc-plugin#33]
**[Rob Dickens](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=lafros)** opened **[MJAVADOC-76](https://issues.apache.org/jira/browse/MJAVADOC-76?redirect=false)** and commented

The summary about says it all. In an earlier project, involving modules, and with aggregate set to true, the contents of doc-files directectories does get copied. Now, where my project does not involve modules, it doesn't.

---
No further details from [MJAVADOC-76](https://issues.apache.org/jira/browse/MJAVADOC-76?redirect=false)

**Comment by olamy** [comment]
**[Nathan Beyer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nbeyer_cerner)** commented

My 'doc-files' folder isn't getting copied either, but I'm only using a very simple and plain project setup (no modules). What I've deduced is that if the 'doc-files' folders are in the source folder (as defined) in the POM, they will get copied. If the 'doc-files' are in any of the resources folders, they will not get copied.

It seems the javadoc tool looks for the 'doc-files' based on what's passed in the "-sourcepath" option and the plugin is setting this to the source folder.

BTW: My environment is Maven 2.0.4, Sun JDK 5.0_7, Windows XP SP2

**Comment by olamy** [comment]
**[Nathan Beyer](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=nbeyer_cerner)** commented

According to the javadoc tool documentation, the 'sourcepath' paramter can be ommited and the source will be searched for in the 'classpath' parameter. Perhaps, the default setup should be to put the sourceDirectory in the 'classpath' parameter if the 'sourcepath' configuration is not set. This way, the source is found for javadoc to process and the doc-files and contents will be in outputDirectory, so the doc-files folder will be found.

**Comment by olamy** [comment]
**[Rob Dickens](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=lafros)** commented

It looks like the bug was confined to the version I had built from src--2.0 as of 2006-05-05. (Version 2.0 wasn't released until 2006-05-14.)

I just cleared this from my local repository, thereby forcing the released version to be downloaded, and the problem went away.

As a test, I rebuilt and installed the plugin from src once again, just to check that the problem came back. It didn't! However, certain things got downloaded this time.

Anyway, sorry for causing unnecessary concern.

Rob
